compiler/ztests/pushdown.yaml

```diff
@@ -3,12 +3,18 @@ script: |
   super compile -C -O 'from file | debug a | values b'
   echo === distinct
   super compile -C -O 'from file | distinct a | values b'
-  echo === fork-join
+  echo === fork into hash join
   super compile -C -O 'fork ( from file1 ) ( from file2 ) | join on right.a=left.b | values left.c,right.d'
-  echo === join
+  echo === fork into nested loop join
+  super compile -C -O 'fork ( from file1 ) ( from file2 ) | join on right.a>left.b | values left.c,right.d'
+  echo === hash join
   super compile -C -O "from file1 | join (from file2) on left.a=right.b | values left.c,right.d"
-  echo === switch-join
+  echo === nested loop join
+  super compile -C -O "from file1 | join (from file2) on left.a>right.b | values left.c,right.d"
+  echo === switch into hash join
   super compile -C -O 'from file | switch a case b ( put x:=c ) case d ( put x:=e ) | join on left.f=right.g | values left.h,right.i'
+  echo === switch into nested loop join
+  super compile -C -O 'from file | switch a case b ( put x:=c ) case d ( put x:=e ) | join on left.f>right.g | values left.h,right.i'
```

Comment:
I might add one trickier test to make sure a reverse order expression equality expression that is somewhat complicated gets correctly identified and flipped (appears to be working in your pr):
```
select *
from (values (1,2),(2,3)) as t1(a,b)
left join (values (2),(3)) as t2(c) on c = a+b
```